pyleoclim/tests/test_utils_tsutils.py

```diff
@@ -0,0 +1,170 @@
+
+
+def test_time_unit_to_datum_exp_dir_unknown_time_unit():
+    # unknown time unit, gets default values
+    time_unit = 'unknown_value'
+    with pytest.warns(match='Time unit'):
+        (datum, exponent, direction) = tsutils.time_unit_to_datum_exp_dir(time_unit)
+        assert datum == 0
+        assert exponent == 0
+        assert direction == 'prograde'
+
+    with pytest.warns(match='Time unit'):
+        (datum, exponent, direction) = tsutils.time_unit_to_datum_exp_dir(time_unit, time_name='unknown')
+        assert direction == 'prograde'
+
+    with pytest.warns(match='Time unit'):
+        (datum, exponent, direction) = tsutils.time_unit_to_datum_exp_dir(time_unit, time_name='age')
+        assert direction == 'retrograde'   
+
+
+@pytest.mark.parametrize('time_unit', tsutils.MATCH_A)
+def test_time_unit_to_datum_exp_dir_yr(time_unit):
+
+    (datum, exponent, direction) = tsutils.time_unit_to_datum_exp_dir(time_unit)
+    assert datum == 0
+    assert exponent == 0
+    assert direction == 'prograde'
+
+
+@pytest.mark.parametrize('time_unit', tsutils.MATCH_KA)
+def test_time_unit_to_datum_exp_dir_ka(time_unit):
+    (datum, exponent, direction) = tsutils.time_unit_to_datum_exp_dir(time_unit)
+    assert datum == 1950
+    assert exponent == 3
+    assert direction == 'retrograde'
+
+
+@pytest.mark.parametrize('time_unit', tsutils.MATCH_MA)
+def test_time_unit_to_datum_exp_dir_ma(time_unit):
+    (datum, exponent, direction) = tsutils.time_unit_to_datum_exp_dir(time_unit)
+    assert datum == 1950
+    assert exponent == 6
+    assert direction == 'retrograde'
+
+
+@pytest.mark.parametrize('time_unit', tsutils.MATCH_GA)
+def test_time_unit_to_datum_exp_dir_ga(time_unit):
+    (datum, exponent, direction) = tsutils.time_unit_to_datum_exp_dir(time_unit)
+    assert datum == 1950
+    assert exponent == 9
+    assert direction == 'retrograde'
+
+
+@pytest.mark.parametrize('time_unit', ['yr BP', 'yrs BP', 'years BP'])
+def test_time_unit_to_datum_exp_dir_yrbp(time_unit):
+    (datum, exponent, direction) = tsutils.time_unit_to_datum_exp_dir(time_unit)
+    assert datum == 1950
+    assert exponent == 0
+    assert direction == 'retrograde'
+
+
+def test_time_unit_to_datum_exp_dir_b2k():
+    time_unit = 'b2k'
+    (datum, exponent, direction) = tsutils.time_unit_to_datum_exp_dir(time_unit)
+    assert datum == 2000
+    assert exponent == 0
+    assert direction == 'retrograde' 
+
+
+@pytest.mark.parametrize('time_unit', ['bp', 'bnf'])
+def test_time_unit_to_datum_exp_dir_bp(time_unit):
+    (datum, exponent, direction) = tsutils.time_unit_to_datum_exp_dir(time_unit)
+    assert datum == 1950
+    assert exponent == 0
+    assert direction == 'retrograde'
+
+
+@pytest.mark.parametrize('time_unit', ['AD', 'CE'])
+def test_time_unit_to_datum_exp_dir_ad(time_unit):
+    (datum, exponent, direction) = tsutils.time_unit_to_datum_exp_dir(time_unit)
+    assert datum == 0
+    assert exponent == 0
+    assert direction == 'prograde'
+
+
```

Comment:
@CommonClimate @khider What I could really use your help on is to make sure the following 6 tests with `expected` values are correct. Are the expected arrays correct?

Reply:
What is `dataframe_dt`? I cannot figure out if the `expected` output is correct because I don't understand what the input is supposed to be.

Reply:
It's defined in `pyleoclim/tests/conftest.py`:

```
@pytest.fixture
def dataframe_dt():
    """Pandas dataframe with a datetime index and random values"""
    length = 5
    dti = pd.date_range("2018-01-01", periods=length, freq="Y")
    df = pd.DataFrame(np.array(range(length)), index=dti)
    return df
```

It's just a way to create an object which you can re-use in different tests, rather than having to write
```
    length = 5
    dti = pd.date_range("2018-01-01", periods=length, freq="Y")
    dataframe_dt = pd.DataFrame(np.array(range(length)), index=dti)
```
at the beginning of every test

https://youtu.be/ScEQRKwUePI

Reply:
Hadn't seen that. Makes a ton of sense, and there are probably of lot of opportunities to improve our existing pytests that way!